napari/plugins/utils.py

```diff
@@ -82,8 +83,12 @@ def _get_preferred_readers(path: str) -> Iterable[Tuple[str, str]]:
     filtered_preferences : Iterable[Tuple[str, str]]
         Filtered patterns and their corresponding readers.
     """
-    reader_settings = get_settings().plugins.extension2reader
 
+    if osp.isdir(path):
+        path = osp.basename(path)
+        if not path.endswith('/'):
+            path = path + os.sep
```

Comment:
`os.sep` should be used in both lines
```suggestion
        if not path.endswith(os.sep):
            path = path + os.sep
```